kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ActionsIterator.java

```diff
@@ -179,19 +181,85 @@ private CloseableIterator<ActionWrapper> getNextActionsIter() {
 
                 return combine(dataIter, false /* isFromCheckpoint */, fileVersion);
             } else if (isCheckpointFile(nextFilePath.getName())) {
-                final long fileVersion = checkpointVersion(nextFilePath);
+                CheckpointInstance checkpointInstance =
+                        new CheckpointInstance(nextFilePath.getName());
+                final long fileVersion = checkpointInstance.version;
 
                 // Try to retrieve the remaining checkpoint files (if there are any) and issue
                 // read request for all in one go, so that the {@link ParquetHandler} can do
                 // optimizations like reading multiple files in parallel.
-                CloseableIterator<FileStatus> checkpointFilesIter =
-                        retrieveRemainingCheckpointFiles(nextFile, fileVersion);
-
+                List<FileStatus> checkpointFiles =
```

Comment:
in generical this if block is getting complicated.

can we divide this into two methods: one for handling multi-part checkpoint and other file handling classic (with or without sidecars) and v2 checkpoint?